openwisp_utils/base.py

```diff
@@ -3,23 +3,38 @@
 from django.db import models
 from django.utils.translation import ugettext_lazy as _
 from model_utils.fields import AutoCreatedField, AutoLastModifiedField
+from openwisp_utils.utils import get_random_key
+from openwisp_utils.validators import key_validator
 
 
-class TimeStampedEditableModel(models.Model):
+class UUIDModel(models.Model):
+    id = models.UUIDField(primary_key=True, default=uuid.uuid4, editable=False)
+
+    class Meta:
+        abstract = True
+
+
+class TimeStampedEditableModel(UUIDModel):
     """
     An abstract base class model that provides self-updating
     ``created`` and ``modified`` fields.
     """
-    id = models.UUIDField(primary_key=True, default=uuid.uuid4, editable=False)
     created = AutoCreatedField(_('created'), editable=True)
     modified = AutoLastModifiedField(_('modified'), editable=True)
 
     class Meta:
         abstract = True
 
 
-class UUIDModel(models.Model):
-    id = models.UUIDField(primary_key=True, default=uuid.uuid4, editable=False)
-
-    class Meta:
-        abstract = True
+class KeyField(models.CharField):
+    def __init__(self, *args, **kwargs):
+        fields = {
+            'max_length': 64,
+            'unique': True,
+            'db_index': True,
+            'default': get_random_key,
+            'validators': [key_validator],
+            'help_text': _('unique key')
+        }
+        fields.update(kwargs)
+        super(KeyField, self).__init__(*args, **fields)
```

Comment:
use python 3 way of doing it, `super()` (rebase on the current master if you're still using the python 2 compatible version).